### PR TITLE
Implement --raw for nix-instantiate --eval

### PR DESCRIPTION
**Pull request by not-my-profile**
## Motivation

The experimental `nix eval` command already supports a `--raw` flag.
This commit implements the same flag for the stable nix-instantiate command.

Until now instructions and scripts that didn't want to rely on experimental
features had to use workarounds such as:

    nix-instantiate --eval <something> | tr -d \"

(which also undesirably also removes double quotation marks within the string), or

    nix-instantiate --eval <something> | jq -j

(which undesirably depends on another package).

## Context

Continuation of #9361.

---

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

The Nix maintainer team uses a [GitHub project board](https://github.com/orgs/NixOS/projects/19) to [schedule and track reviews](https://github.com/NixOS/nix/tree/master/maintainers#project-board-protocol).

### CODE REVIEW

**Review by roberth — COMMENTED** [review]
Document the newline behavior, then lgtm!

File: src/nix-instantiate/nix-instantiate.cc [hunk 1]
```diff
@@ -50,7 +50,11 @@ void processExpr(EvalState & state, const Strings & attrPaths,
                 vRes = v;
             else
                 state.autoCallFunction(autoArgs, v, vRes);
-            if (output == okXML)
+            if (output == okRaw)
+                std::cout << *state.coerceToString(noPos, vRes, context, "while generating the nix-instantiate output");
+                // We intentionally don't output a newline here. The default PS1 for Bash in NixOS starts with a newline
```
[the remark this answers is not in the document]
Comment (not-my-profile) [reply src/nix-instantiate/nix-instantiate.cc:55]: I mentioned the absence of a trailing newline in the manual.

**Review by roberth — APPROVED** [review]

**Comment by roberth** [comment]
@mergify queue

**Comment by not-my-profile** [comment]
I think the CI workflow is still awaiting approval.

**Comment by edolstra** [comment]
I'm not super in favor of adding functionality to the legacy CLI that already exists in the new CLI. It does add a bunch of (essentially superfluous) code that we'll need to maintain indefinitely.

**Comment by roberth** [comment]
I hear you. Until we have stabilized the new CLI, users will have unmet needs on the old CLI for use in Nixpkgs etc, which rightfully has a policy of avoiding the experimental CLI. This allows us to apply learned insights to the new CLI without as much risk.
I agree for things like [this](https://github.com/NixOS/nix/pull/12119#discussion_r1900126156) which should arguably be implemented for the new CLI only. Stable CLI users can achieve essentially the same result with `concatStringsSep` and such. The same doesn't apply to `--raw`.

**Comment by not-my-profile** [comment]
(fixed YAML syntax error in release note frontmatter)

**Comment by roberth** [comment]
> I think the CI workflow is still awaiting approval.

Didn't have the button for it until I switched back out of the new commit status view beta! (" :sparkles: Try the new merge experience")

**Pull request merged**